vendor/forms.py

```diff
@@ -46,7 +47,7 @@ class Meta:
 
 
 class OfferForm(forms.ModelForm):
-    products = forms.ModelMultipleChoiceField(label=_("Available Products:"), required=True, queryset=Product.on_site.filter(available=True))
+    products = forms.ModelMultipleChoiceField(label=_("Available Products:"), required=True, queryset=Product.objects.filter(site=settings.SITE_ID, available=True))
```

Comment:
Can not the settings.SITE_ID as it would not be compatible with site-alias approach.

Reply:
That is just a placeholder until the __init__ is called otherwise it would through an error that a query is required for that field. The correct query is being set in the __init__ functions:
```
self.fields['products'].queryset = Product.objects.filter(site=request.site, available=True)
```

Reply:
updated to:
```
self.fields['products'].queryset = Product.objects.filter(site=get_site_from_request(request), available=True)
```